tests/unit/other.tcl

```diff
@@ -39,8 +39,8 @@ start_server {tags {"other"}} {
     }
 
     test {BGSAVE} {
-        waitForBgsave r
         r flushdb
+        waitForBgsave r
         r save
```

Comment:
it look like there is a race condition: https://github.com/redis/redis-extra-ci/runs/6331356284?check_suite_focus=true#step:7:3057
So it look like `r flushdb` trigger (schedule) a bgsave right after `waitForBgsave r` and before `r save`
```
[exception]: Executing test client: ERR Background save already in progress.
ERR Background save already in progress
    while executing
"[srv $level "client"] {*}$args"
    (procedure "r" line 7)
    invoked from within
"r save"
    ("uplevel" body line 4)
    invoked from within
"uplevel 1 $code"
    (procedure "test" line 51)
    invoked from within
"test {BGSAVE} {
        r flushdb
        waitForBgsave r
        r save
        r set x 10
        r bgsave
        waitForBgsave r
        r debug r..."
    ("uplevel" body line 41)
    invoked from within
"uplevel 1 $code "
    (procedure "start_server" line 3)
    invoked from within
"start_server {tags {"other"}} {
    if {$::force_failure} {
        # This is used just for test suite development purposes.
        test {Failing tes..."
    (file "tests/unit/other.tcl" line 1)
    invoked from within
```

Reply:
@enjoy-binbin thank you for noticing this, and also figuring out the problem!
we can't afford using a MULTI sine SAVE is not allowed in MULTI.
but how about changing FLUSHDB to FLUSHALL?
it does actually do a foreground save, and then sets the dirty counter to 1.
wanna test it and make a PR?

Reply:
FLUSHALL won't set the dirty counter to 1
https://github.com/redis/redis/blob/ff3a3577f29a07e965b3b38d07894410d28da499/src/db.c#L600-L624

Reply:
ohh, i missed the fact we restore it, but we do increment it again below.
maybe the right thing would be to just increment it by one.
i.e. remove the increment in `server.dirty += emptyData`, and keep all the other lines (restoring it after `save` zeroed it), and then incrementing by one.

or maybe the right thing would be to remove all that mess (let `save` reset it), but add a call to `forceCommandPropagation`.
that seems like the right thing to do instead of all the messing around with the dirty counter.

Reply:
+1 for `forceCommandPropagation`

Reply:
yes, i will go for `forceCommandPropagation`, and the meantime, it reminds me of this: #8972